aosc-os-core: 8.0.0, "Hotfix"

MingcongBai opened the pull request:
Topic Description
-----------------

General Core 8 component updates, rebuilds, and fixes.

- Update GCC to 10.2.1, Glibc to 2.32, etc. Rebuild all static libraries.
- Update Perl to 5.32, rebuild all reverse dependencies.
- Downgrade Berkeley DB to license-compliant version 5, rebuild all reverse dependencies.
- Introduce `libnsl2`, `rpcsvc-proto` compatibility libraries. Adjust dependencies where applicable.

Package(s) Affected
-------------------

- `groups/core`
- `groups/db-rebuilds`
- `groups/libnsl2-rebuilds`
- `groups/static-rebuilds`
- `groups/perl-rebuilds`
- `autobuild3` v20201124
- `camlp5` v713
- `dune` v2.7.1
- `erlang` v23.1
- `flint` v2.5.2-8
- `gnome-doc-utils` v0.20.10-3
- `lablgtk` v2.18.11
- `libappindicator` v12.10.0-4
- `libnih` v1.0.3-2
- `lrs` v0.71a
- `ocaml` v4.09.1
- `ocaml-ulex` v1.2
- `ocamlgraph` v2.0.0
- `tqt3` v14.0.7-2
- `trinity-libkdcraw` v14.0.7-1
- `weechat` v3.0
- `xapian-core` v1.4.17 (rebuild with gcc 10 before groups/perl-rebuilds)
- `zxing-cpp` v1.1.1

### Additions

- `libnsl2` v1.3.0
- `rpcsvc-proto` v1.4.2
- `ocaml-csexp` v1.3.2
- `ocaml-result` v1.5
- `ocaml-graphics` v5.1.0
- `ocaml-stdlib-shims` v0.2.0
- `perl-text-soundex` v3.05

### Removals

- Unmaintainable packages...
    - `pfff`, `hhvm`
- Perl 5.32 CPAN mergers...
   - `perl-carp`, `perl-if`, `perl-exporter`, `perl-digest-sha`, `perl-xsloader`,` perl-mime-base64`, `perl-text-balanced`

Security Update?
----------------

No

Build Order
-----------

```
groups/core
autobuild3
perl-locale-gettext
perl-xml-parser
gtest
libnsl2
groups/db-rebuilds
rpcsvc-proto
tcl
libnih
groups/libnsl2-rebuilds
groups/static-rebuilds
xapian-core
groups/perl-rebuilds
gnome-doc-utils
libappindicator
xapian-core
flint
tqt3
trinity-libkdcraw
```

Architectural Progress
----------------------

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

Secondary Architectural Progress
--------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

----

After the pull request is merged, all package(s) affected must be rebuilt against the `stable` Git tree and environment (only `stable` repository should be enabled in `sources.list`). This section marks the progress above.

Please, make sure the list of architectures below matches the ones above.

Post-Merge Architectural Progress
---------------------------------

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

Post-Merge Secondary Architectural Progress
-------------------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

MingcongBai committed:
perl-math-random-isaac: rebuild for Perl 5.32.0

MingcongBai committed:
perl-module-scandeps: rebuild for Perl 5.32.0

MingcongBai committed:
perl-mime-types: rebuild for Perl 5.32.0

MingcongBai committed:
perl-mime-tools: rebuild for Perl 5.32.0

MingcongBai committed:
perl-module-build-tiny: rebuild for Perl 5.32.0

MingcongBai committed:
perl-mro-compat: rebuild for Perl 5.32.0

MingcongBai committed:
perl-module-find: rebuild for Perl 5.32.0

MingcongBai committed:
perl-module-pluggable: rebuild for Perl 5.32.0

MingcongBai committed:
perl-namespace-autoclean: rebuild for Perl 5.32.0

MingcongBai committed:
perl-module-implementation: rebuild for Perl 5.32.0

MingcongBai committed:
perl-mozilla-ca: rebuild for Perl 5.32.0

MingcongBai committed:
perl-module-build: rebuild for Perl 5.32.0

MingcongBai committed:
perl-mime-lite: rebuild for Perl 5.32.0

MingcongBai committed:
perl-mime-base64: rebuild for Perl 5.32.0

MingcongBai committed:
perl-module-runtime: rebuild for Perl 5.32.0

MingcongBai committed:
perl-netaddr-ip: rebuild for Perl 5.32.0

MingcongBai committed:
perl-pango: rebuild for Perl 5.32.0

MingcongBai committed:
perl-package-stash: rebuild for Perl 5.32.0

MingcongBai committed:
perl-net-smtp-ssl: rebuild for Perl 5.32.0

MingcongBai committed:
perl-namespace-clean: rebuild for Perl 5.32.0

MingcongBai committed:
perl-net-server: rebuild for Perl 5.32.0

MingcongBai committed:
perl-package-stash-xs: rebuild for Perl 5.32.0

MingcongBai committed:
perl-ntlm: rebuild for Perl 5.32.0

MingcongBai committed:
perl-net-dns-resolver-programmable: rebuild for Perl 5.32.0

MingcongBai committed:
perl-net-ip: rebuild for Perl 5.32.0

MingcongBai committed:
perl-net-http: rebuild for Perl 5.32.0

MingcongBai committed:
perl-net-dbus: rebuild for Perl 5.32.0

MingcongBai committed:
perl-net-ssleay: rebuild for Perl 5.32.0

MingcongBai committed:
perl-net-dns: rebuild for Perl 5.32.0

MingcongBai committed:
perl-patchreader: rebuild for Perl 5.32.0

MingcongBai committed:
perl-parse-recdescent: rebuild for Perl 5.32.0

MingcongBai committed:
perl-par: rebuild for Perl 5.32.0

MingcongBai committed:
perl-params-validationcompiler: rebuild for Perl 5.32.0

MingcongBai committed:
perl-params-util: rebuild for Perl 5.32.0

MingcongBai committed:
perl-par-dist: rebuild for Perl 5.32.0

MingcongBai committed:
perl-path-tiny: rebuild for Perl 5.32.0

MingcongBai committed:
perl-proc-processtable: rebuild for Perl 5.32.0

MingcongBai committed:
perl-pod-coverage: rebuild for Perl 5.32.0

MingcongBai committed:
perl-path-class: rebuild for Perl 5.32.0

MingcongBai committed:
perl-par-packer: rebuild for Perl 5.32.0

MingcongBai committed:
perl-proc-simple: rebuild for Perl 5.32.0

MingcongBai committed:
perl-parallel-forkmanager: rebuild for Perl 5.32.0

MingcongBai committed:
perl-socket6: rebuild for Perl 5.32.0

MingcongBai committed:
perl-sort-versions: rebuild for Perl 5.32.0

MingcongBai committed:
perl-role-tiny: rebuild for Perl 5.32.0

MingcongBai committed:
perl-string-util: rebuild for Perl 5.32.0

MingcongBai committed:
perl-specio: rebuild for Perl 5.32.0

MingcongBai committed:
perl-sgmlspm: rebuild for Perl 5.32.0

MingcongBai committed:
perl-soap-lite: rebuild for Perl 5.32.0

MingcongBai committed:
perl-sort-naturally: rebuild for Perl 5.32.0

MingcongBai committed:
perl-return-value: rebuild for Perl 5.32.0

MingcongBai committed:
perl-string-shellquote: rebuild for Perl 5.32.0

MingcongBai committed:
perl-regexp-common: rebuild for Perl 5.32.0

MingcongBai committed:
perl-rpc-xml: rebuild for Perl 5.32.0

MingcongBai committed:
perl-regexp-assemble: rebuild for Perl 5.32.0

MingcongBai committed:
perl-readonly: rebuild for Perl 5.32.0

MingcongBai committed:
perl-term-readkey: rebuild for Perl 5.32.0

MingcongBai committed:
perl-term-readline-gnu: rebuild for Perl 5.32.0

MingcongBai committed:
perl-sub-install: rebuild for Perl 5.32.0

MingcongBai committed:
perl-test-deep: rebuild for Perl 5.32.0

MingcongBai committed:
perl-sys-virt: rebuild for Perl 5.32.0

MingcongBai committed:
perl-template-gd: rebuild for Perl 5.32.0

MingcongBai committed:
perl-term-progressbar: rebuild for Perl 5.32.0

MingcongBai committed:
perl-template-toolkit: rebuild for Perl 5.32.0

MingcongBai committed:
perl-sub-exporter-progressive: rebuild for Perl 5.32.0

MingcongBai committed:
perl-sys-meminfo: rebuild for Perl 5.32.0

MingcongBai committed:
perl-test-mock-guard: rebuild for Perl 5.32.0

MingcongBai committed:
perl-sub-quote: rebuild for Perl 5.32.0

MingcongBai committed:
perl-sub-identify: rebuild for Perl 5.32.0

MingcongBai committed:
perl-test-fatal: rebuild for Perl 5.32.0

MingcongBai committed:
perl-timedate: rebuild for Perl 5.32.0

MingcongBai committed:
perl-test-pod: rebuild for Perl 5.32.0

MingcongBai committed:
perl-test-requires: rebuild for Perl 5.32.0

MingcongBai committed:
perl-text-balanced: rebuild for Perl 5.32.0

MingcongBai committed:
perl-test-pod-coverage: rebuild for Perl 5.32.0

MingcongBai committed:
perl-text-csv: rebuild for Perl 5.32.0

MingcongBai committed:
perl-text-iconv: rebuild for Perl 5.32.0

MingcongBai committed:
perl-text-csv-xs: rebuild for Perl 5.32.0

MingcongBai committed:
perl-text-wrapi18n: rebuild for Perl 5.32.0

MingcongBai committed:
perl-text-charwidth: rebuild for Perl 5.32.0

MingcongBai committed:
perl-test-nowarnings: rebuild for Perl 5.32.0

MingcongBai committed:
perl-text-template: rebuild for Perl 5.32.0

MingcongBai committed:
perl-time-duration: rebuild for Perl 5.32.0

MingcongBai committed:
perl-test-mockobject: rebuild for Perl 5.32.0

MingcongBai committed:
perl-type-serialiser: rebuild for Perl 5.32.0

MingcongBai committed:
perl-x11-protocol-other: rebuild for Perl 5.32.0

MingcongBai committed:
perl-universal-can: rebuild for Perl 5.32.0

MingcongBai committed:
perl-www-curl: rebuild for Perl 5.32.0

MingcongBai committed:
perl-universal-isa: rebuild for Perl 5.32.0

MingcongBai committed:
perl-uri: rebuild for Perl 5.32.0

MingcongBai committed:
perl-tk: rebuild for Perl 5.32.0

MingcongBai committed:
perl-xml-filter-buffertext: rebuild for Perl 5.32.0

MingcongBai committed:
perl-www-mechanize: rebuild for Perl 5.32.0

MingcongBai committed:
perl-www-robotrules: rebuild for Perl 5.32.0

MingcongBai committed:
perl-xml-libxml: rebuild for Perl 5.32.0

MingcongBai committed:
perl-try-tiny: rebuild for Perl 5.32.0

MingcongBai committed:
perl-x11-protocol: rebuild for Perl 5.32.0

MingcongBai committed:
perl-unicode-string: rebuild for Perl 5.32.0

MingcongBai committed:
perl-xsloader: rebuild for Perl 5.32.0

MingcongBai committed:
perl-xml-xpath: rebuild for Perl 5.32.0

MingcongBai committed:
gnumeric: rebuild for Perl 5.32.0

MingcongBai committed:
perl-xml-namespacesupport: rebuild for Perl 5.32.0

MingcongBai committed:
perl-xml-sax: rebuild for Perl 5.32.0

MingcongBai committed:
perl-xml-sax-expat: rebuild for Perl 5.32.0

MingcongBai committed:
perl-xml-sax-base: rebuild for Perl 5.32.0

MingcongBai committed:
polymake: rebuild for Perl 5.32.0

MingcongBai committed:
perl4-corelibs: rebuild for Perl 5.32.0

MingcongBai committed:
perl-xml-simple: rebuild for Perl 5.32.0

MingcongBai committed:
perl-xml-writer: rebuild for Perl 5.32.0

MingcongBai committed:
perl-xml-libxslt: rebuild for Perl 5.32.0

MingcongBai committed:
perl-xml-sax-writer: rebuild for Perl 5.32.0

MingcongBai committed:
perl-xml-twig: rebuild for Perl 5.32.0

MingcongBai committed:
znc: rebuild for Perl 5.32.0

MingcongBai committed:
imagemagick: rebuild for Perl 5.32.0

MingcongBai committed:
pcsc-perl: rebuild for Perl 5.32.0

MingcongBai committed:
subversion: rebuild for Perl 5.32.0

MingcongBai committed:
irssi: rebuild for Perl 5.32.0

MingcongBai committed:
rxvt-unicode: rebuild for Perl 5.32.0

MingcongBai committed:
claws-mail: rebuild for Perl 5.32.0

MingcongBai committed:
openresty: rebuild for Perl 5.32.0

MingcongBai committed:
weechat: rebuild for Perl 5.32.0

MingcongBai committed:
xfconf: rebuild for Perl 5.32.0

MingcongBai committed:
tdebindings: rebuild for Perl 5.32.0

MingcongBai committed:
pcp: rebuild for Perl 5.32.0

MingcongBai committed:
hexchat: rebuild for Perl 5.32.0

MingcongBai committed:
kvirc: rebuild for Perl 5.32.0

MingcongBai committed:
groups/perl-rebuilds: adjust for Core 8

MingcongBai committed:
kernel-tools: (upstream patch) fix build with GCC 10

MingcongBai committed:
perl-async-interrupt: update to 1.26; add perl-canary-stability build dep

MingcongBai committed:
perl-carp: drop, now part of perl

MingcongBai committed:
perl-cgi: drop perl-if dep

MingcongBai committed:
perl-exporter: drop, now part of perl

MingcongBai committed:
perl-if: drop, now part of perl

MingcongBai committed:
perl-digest-sha: drop, now part of perl

MingcongBai committed:
perl-xsloader: drop, now part of perl

MingcongBai committed:
perl-mime-base64: drop, now part of perl

MingcongBai committed:
perl-text-balanced: drop, now part of perl

MingcongBai committed:
perl-extutils-helpers: drop perl-carp, perl-exporter deps

MingcongBai committed:
perl-extutils-installpaths: drop perl-carp dep

MingcongBai committed:
perl-io-tee: drop perl-carp dep

MingcongBai committed:
perl-json-webtoken: drop perl-carp, perl-exporter, perl-mime-base64, perl-digest-sha deps

MingcongBai committed:
perl-ldap: add perl-text-soundex dep

MingcongBai committed:
perl-mail-imapclient: drop perl-mime-base64 dep

MingcongBai committed:
perl-module-build-tiny: drop perl-exporter dep

MingcongBai committed:
perl-ntlm: drop perl-mime-base64 dep

MingcongBai committed:
perl-packages-stash-xs: drop perl-xsloader build dep

MingcongBai committed:
perl-parse-recdescent: drop perl-text-balanced dep

MingcongBai committed:
perl-test-fatal: drop perl-carp, perl-exporter deps

MingcongBai committed:
perl-unicode-string: drop perl-mime-base64 dep

MingcongBai committed:
perl-www-curl: (Fedora patches) fix build against cURL >= 7.69.0

MingcongBai committed:
subversion: disable RECONF, update conffiles

MingcongBai committed:
openresty: drop RECONF

MingcongBai committed:
weechat: update to 3.0

MingcongBai committed:
znc: drop Git submodule operations

MingcongBai committed:
perl-text-soundtex: new, 3.05

MingcongBai committed:
kvirc: (upstream patch) fix build with Qt 5.15

MingcongBai committed:
groups/perl-rebuilds: drop deprecated packages

MingcongBai committed:
perl: update breakage and replacement

MingcongBai committed:
perl: fix breakage versions

Icenowy reviewed (approved):
Looks good to me, altough I don't have time to check each individual rebuild.

BTW how should we deal with the rebuild of this huge change? Lock down master during the rebuild?

MingcongBai commented:
> BTW how should we deal with the rebuild of this huge change? Lock down master during the rebuild?

The bigger issue is to control a (relatively) clean environment. I wouldn't worry too much about `stable`, since we have full control over what goes in and what doesn't (yet).

MingcongBai committed:
linux-pam: add libnsl2 dep

MingcongBai committed:
netbeans: add libnsl2 dep

MingcongBai committed:
gmime-3: add libnsl2 dep

MingcongBai committed:
fcgi: add libnsl2 dep

MingcongBai committed:
graphviz: add libnsl2 dep

MingcongBai committed:
gmime: add libnsl2 dep

MingcongBai committed:
opensp: add libnsl2 dep

MingcongBai committed:
python-3: add libnsl2 dep

MingcongBai committed:
python-2: add libnsl2 dep

MingcongBai committed:
autofs: add libnsl2 dep

MingcongBai committed:
libasyncns: add libnsl2 dep

MingcongBai committed:
php7: add libnsl2 dep

MingcongBai committed:
autossh: add libnsl2 dep

MingcongBai committed:
dovecot: add libnsl2 dep

MingcongBai committed:
xine-lib: add libnsl2 dep

MingcongBai committed:
bind: add libnsl2 dep

MingcongBai committed:
exim: add libnsl2 dep

MingcongBai committed:
loudmouth: add libnsl2 dep

MingcongBai committed:
samba: add libnsl2 dep

MingcongBai committed:
xerces-c: add libnsl2 dep

MingcongBai committed:
netease-cloud-music: add libnsl2 dep

MingcongBai committed:
net-snmp: add libnsl2 dep

MingcongBai committed:
freeradius-client: add libnsl2 dep

MingcongBai committed:
openslp: add libnsl2 dep

MingcongBai committed:
libuv: add libnsl2 dep

MingcongBai committed:
tcp-wrappers: add libnsl2 dep

MingcongBai committed:
pcp: add libnsl2 dep

MingcongBai committed:
squid: add libnsl2 dep

MingcongBai committed:
stackless-3: add libnsl2 dep

MingcongBai committed:
stunnel: add libnsl2 dep

MingcongBai committed:
claws-mail: add libnsl2 dep

MingcongBai committed:
stackless-2: add libnsl2 dep

MingcongBai committed:
aria2: add libnsl2 dep

MingcongBai committed:
gnumeric: add libnsl2 dep

MingcongBai committed:
sylpheed: add libnsl2 dep

MingcongBai committed:
pidgin: add libnsl2 dep

MingcongBai committed:
aosc-aaa: fix postinst pattern

MingcongBai committed:
iproute2: update conffiles

MingcongBai committed:
python-2: disable x86 intrinsics for ppc64el

MingcongBai committed:
python-3: disable x86 intrinsics for ppc64el

MingcongBai committed:
tcc: only builds on amd64, arm64

MingcongBai committed:
camlp5: update to 713 to fix build with OCaml 4.09

MingcongBai committed:
numpy: lint script; disable x86 intrinsics for ppc64el

MingcongBai committed:
lapack: disable x86 intrinsics for ppc64el

MingcongBai committed:
geany: disable LTO for loongson3

MingcongBai committed:
cpp-hocon: disable LTO for loongson3

MingcongBai committed:
xerces-c: disable LTO for loongson3

MingcongBai committed:
squid: disable LTO for loongson3

MingcongBai committed:
mysql-connector-c++: disable LTO for loongson3

MingcongBai committed:
leatherman: disable LTO for loongson3

MingcongBai committed:
gnome-doc-utils: disable RECONF

MingcongBai committed:
flint: use self ABTYPE

MingcongBai committed:
libappindicator: append -fcommon to fix build

MingcongBai committed:
singular: disable strict Autotools option checking

MingcongBai committed:
xapian-core: rebuild using GCC 10

- On ppc64el, if xapian-core was built using GCC 9, an error
  occurs when building xapian-bindings.

MingcongBai committed:
autobuild3: update to 20201202 (amd64 SSE2 adjustment)

MingcongBai committed:
gringo: disable LTO for loongson3

MingcongBai committed:
gdal: disable MongoDB support for loongson3

MingcongBai committed:
tachyon: fix build on loongson3

MingcongBai committed:
libnih: fix build with GCC 10

MingcongBai committed:
xapian-bindings: disable C# support for loongson3

MingcongBai committed:
xapian-core: disable parallelism

MingcongBai committed:
mono: does not build on !(amd64|arm64|ppc64el)

MingcongBai committed:
camlp4: limit native arch to amd64, arm64, ppc64el

MingcongBai committed:
camlp5: limit native arch to amd64, arm64, ppc64el

MingcongBai committed:
facile: limit native arch to amd64, arm64, ppc64el

MingcongBai committed:
findlib: limit native arch to amd64, arm64, ppc64el

MingcongBai committed:
lablgtk: limit native arch to amd64, arm64, ppc64el; drop duplicate ld.conf

MingcongBai committed:
ocaml-ulex: limit native arch to amd64, arm64, ppc64el

MingcongBai committed:
ocaml-zip: limit native arch to amd64, arm64, ppc64el; provide camlzip

MingcongBai committed:
ocaml: limit native arch to amd64, arm64, ppc64el

MingcongBai committed:
ocamlbuild: limit native arch to amd64, arm64, ppc64el

MingcongBai committed:
openblas: fix loongson3 support; unify build scripts

MingcongBai committed:
tdegames: disable LTO for loongson3

MingcongBai committed:
tdebase: disable LTO for loongson3

MingcongBai committed:
tdeedu: disable LTO for loongson3

MingcongBai committed:
tdeaddons: disable LTO for loongson3

MingcongBai committed:
tdelibs: disable LTO for loongson3

MingcongBai committed:
tdegraphics: disable LTO for loongson3

MingcongBai committed:
tdesdk: disable LTO for loongson3

MingcongBai committed:
tdeutils: disable LTO for loongson3

MingcongBai committed:
tdevelop: disable LTO for loongson3

MingcongBai committed:
trinity-libkdcraw: disable RECONF

MingcongBai committed:
tdebindings: disable C# support for loongson3

MingcongBai committed:
tdenetwork: disable LTO for loongson3; fix build with GCC 10

Co-authored-by: liushuyu <liushuyu011@gmail.com>

MingcongBai committed:
tdepim: disable LTO for loongson3; use cmake ABTYPE; fix build with GCC 10

Co-authored-by: liushuyu <liushuyu011@gmail.com>

MingcongBai committed:
tqt3: disable LTO for loongson3; fix build with Glibc >= 2.32

MingcongBai committed:
lablgtk: add findlib build dep

MingcongBai committed:
Revert "lablgtk: add findlib build dep"

This reverts commit 215fafe33d92be0b6a690656b7329cc7477ab9d8.

MingcongBai committed:
lablgtk: fix build

MingcongBai committed:
kernel-tools: drop old patch for GCC 10
